### ISSUE TITLE
[BUG] `free(): double free detected in tcache 2` on restore

### ISSUE THREAD

**Issue by flanger001** [mydumper/mydumper#725]
**Describe the bug**
When I run `myloader` on a backup directory, it restores all data successfully but then segfaults when restoring constraints. It emits the error in the title. The attached log file will indicate where this is happening.

**To Reproduce**
Command executed:

```
myloader \
--host="$address" \
--user="$username" \
--password="$password" \
--database="$database_name" \
--directory="$directory_name" \
--no-data \
--innodb-optimize-keys=AFTER_IMPORT_ALL_TABLES \
--threads="$(nproc)" \
--max-threads-per-table="$(nproc)" \
--verbose=3
```

Note the `--no-data` option was so that I could reproduce this successfully and not have it take 6 hours.

**Myloader version**: 
`myloader 0.12.3-3, built against MySQL 8.0.28-19`

**Expected behavior**
I expected the restore to complete successfully with no errors.

**Log**
[restore.log](https://github.com/mydumper/mydumper/files/8900513/restore.log)

**Backup**
Not applicable

**How to repeat**
Running the above command on a backup directory will result in a segfault every time.

Files for reproduction steps are tough because the schema is from a production database. 🙁 I can email you the structure files privately if that would be possible?

**Environment (please complete the following information):**
 - OS version: Debian GNU/Linux 11
 - MyDumper version: `mydumper 0.12.3-3, built against MySQL 8.0.28-19`

**Additional context**
This is being run from a script in a Kubernetes pod, and the database backup directory is on a mounted block volume. The database being restored to is not on the same machine, but it is in the same Kubernetes cluster.

I believe this error is similar to these:

- https://github.com/mydumper/mydumper/issues/703
- https://github.com/mydumper/mydumper/issues/594

Thank you!

**Comment by flanger001** [comment]
I just tried this again with `mydumper 0.12.3-1, built against MySQL 8.0.27-18` and it completed successfully, so I am guessing whatever I am seeing here showed up in 0.12.3-2 or 0.12.3-3.

**Comment by flanger001** [comment]
And @davidducos if you want the structure files to reproduce this with, let me know and I can get them to you.

**Comment by davidducos** [comment]
I created a test for master and it didn't fail. Can you test with master? you will need to compile it locally...

**Comment by davidducos** [comment]
@flanger001 It has been fixed on v0.12.5-1

**Comment by flanger001** [comment]
@davidducos I can attempt this later today for sure. Thank you!
